### PR TITLE
LDA+U Support

### PR DESCRIPTION
**Pull request by janssenhenning**
- Add Support for initializing or modifying a ``n_mmp_mat`` file via the FleurinpModifier.  Also implemented limited validation of the ``n_mmp_mat`` file

- Add ``n_mmp_mat`` file to retrieve_list and copy it for back up for new calculations if it exists in the retrieved folder and no file is in the fleurinpdata provided

- Added 'ldaUinfo' key to simple_data output

- Updated documentation

Things I'm not sure about:

- set_nmmpmat can not be called with complex numbers in the density matrix specification, i.e. ```occStates``` or ```denmat``` (aiida calcfunction cant handle complex numbers as arguments. ```clean_value``` fails)

- Should the final density matrix be parsed into simple_data from the out.xml (Problem: Can complex numbers enter this dictionary)

- Restarts e.g in the scf workchain with a specified n_mmp_mat will reset to the initial guess of n_mmp_mat

### CODE REVIEW

**Comment by broeder-j** [comment]
Sorry @janssenhenning I missed this. Next time please assign us as a reviewer. Then we get more notification.

**Comment by janssenhenning** [comment]
The n_mmp_mat is now removed from the local_copy_list before adding any files from the retrieved folder if the retrieved folder has a ```n_mmp_mat```/```n_mmp_mat_out``` file. This solves the problem of resetting the density matrix within the scf workchain

**Comment by broeder-j** [comment]
Comment: Lines 44, 46, 57, 71, 76-80, 94, 111, 115, 218, 231, 245-257, 263-266 of 'set_nmmpmat.py' are not covered by the tests.

I added a test with phi=pi/2 to run through most of these lines.
left with 47, 49, 60, 74, 80, 97, 114, 221, 234, which is ok and contains mostly errors.
Maybe it makes sense to adapt this test to a something where phi is something non trivial. So far I have seen no difference in the result for phi=pi, pi/2 and phi=0 is this correct behavior?

**Comment by janssenhenning** [comment]
@broeder-j I have to check this but I think this should be correct. If you look at the definition of the d-wigner matrix ```theta``` corresponds to the euler angle ```beta``` and ```phi``` to ```alpha```. For beta = 0 the definition will always collapse into the identity.

So ```phi``` only has an effect for ```theta != 0```

**Review by broeder-j — APPROVED** [review]
Very nice job! Thanks for updating the docs and the tests.

two things:
1. phi=!0 testcase
2. keys in output nodes and input nodes should be lowercase with underscore (Convention so far).

**Comment by broeder-j** [comment]
> @broeder-j I have to check this but I think this should be correct. If you look at the definition of the d-wigner matrix `theta` corresponds to the euler angle `beta` and `phi` to `alpha`. For beta = 0 the definition will always collapse into the identity.
> 
> So `phi` only has an effect for `theta != 0`
> 
> Edit: Correction the rotation matrix will have phases resulting from `phi` on the diagonal but these will cancel in the rotation

Then maybe we should do a theta!=0 test case, or?

File: aiida_fleur/parsers/fleur.py [hunk 1]
```diff
@@ -934,6 +974,15 @@ def write_simple_outnode(value, value_type, value_name, dict_out):
 
                 # Total charges, total magentic moment
 
+            if ldaU:
+                simple_data['ldaUinfo'] = {}
+                eldau = eval_xpath(iteration_node, eldau_xpath)
+                write_simple_outnode(eldau, 'float', 'ldaU_energy_correction', simple_data['ldaUinfo'])
```
[the remark this answers is not in the document]
Comment (broeder-j) [reply aiida_fleur/parsers/fleur.py:980]: mapping of the exact fleur keys (inp.xml dict) is excepted from this.

**Comment by janssenhenning** [comment]
> > @broeder-j I have to check this but I think this should be correct. If you look at the definition of the d-wigner matrix `theta` corresponds to the euler angle `beta` and `phi` to `alpha`. For beta = 0 the definition will always collapse into the identity.
> > So `phi` only has an effect for `theta != 0`
> > Edit: Correction the rotation matrix will have phases resulting from `phi` on the diagonal but these will cancel in the rotation
> 
> Then maybe we should do a theta!=0 test case, or?

Yes I will do that.

File: aiida_fleur/parsers/fleur.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (janssenhenning) [reply aiida_fleur/parsers/fleur.py]: Done with 9f968c8

**Comment by broeder-j** [comment]
@janssenhenning Thank you!

**Pull request merged**